### PR TITLE
Chaos Mesh Project Security Self-Assessment Security Pals

### PR DESCRIPTION
**Pull request by mathur-14**
Created and added first draft for **Chaos Mesh** Project Security Self-Assessment.

Please feel free to share your feedback on the security self-assessment.

### CODE REVIEW

**Comment by eddie-knight** [comment]
Hi there! I'm just getting started looking at your pull request, and I noticed the DCO check is failing.

You can look at the checks section of the PR (I believe it should always be below the last comment) and look for a red X highlighting the failed check. In this case, you can click `Details` for more information about how to get that check passing.
Screenshot 2023-12-08 at 8 35 18 AM

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:1]:
```suggestion
# Chaos Mesh Self-assessment
```

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:11]:
```suggestion
```

Comment (eddie-knight) [reply assessments/projects/chaos-mesh/self-assessment.md:11]: Please go back to ensure that no template text like this is included here or in the rest of the document

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: I've made the required changes.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: I've made the required changes. No pre-texts now exists.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:31]:
```suggestion
| SBOM              | Chaos Mesh automatically generates SBOMs on release: https://github.com/chaos-mesh/chaos-mesh/releases/latest |
```

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:87]: Did the Chaos Mesh team participate in this activity?

Consider naming the authors or contributors in the first paragraph, right below the page header.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:270]: Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document. You may optionally communicate recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md:87]: We did connect with the maintainers of chaos mesh and exchanged information about the project and what this document is about. We did ask them to have a look and share their feedback but so far they haven't got back to us on that.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: Sure, we have removed it from the doc and we'll be moving forward with any recommendations from their side and add them to GitHub issues.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md:92]: We have added the authors and contributors/reviewers at the start.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:39]: This space is also meant to include their vulnerability disclosure, incident response processes, default configurations, and anything else that can supplement security considerations. Please revisit and include them. In the event they aren't available, kindly mark them as not available

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:58]: I see chaos dashboard and tools within containers enabling chaos mesh deliver its full functionalities. 
It maybe worthwhile to include them if they align the definition of actors per https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md#actors

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:65]: It is worthwhile to include if there are any checks performed in each of these actions.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:73]: Is this action coming from the chaos controller manager itself or there is some other component who is involed? Per the architecture, I assume it is the controller manager. Please clarify

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:99]: Please move this threat model to a separate file in the same folder, and link it to the appendix section

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:224]: Please add reference if this is documented anywhere. 
This looks largely manual efforts. Are there any automated tests or established devops/devsecops pipeline in the project?

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:257]: Please provide reference of this process in the appendix

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:278]: How was it determined they are doing well? 
Do they have a badge available for their project?

**Review by ragashreeshekar — CHANGES_REQUESTED** [review]
Thanks for the PR @garethr and team, appreciate the efforts.
I have completed first pass of review and left several comments on section that needs your attention. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (iamgeeyuh) [reply assessments/projects/chaos-mesh/self-assessment.md:73]: Yes the action is coming from the chaos controller manager. I've updated the formatting of the information to make this more apparent

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:278]: We edited the language to clarify that, Chaos mesh does not possess the badge. We analyzed independently using the CII best practices handbook and edited the document that clarifies this.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:99]: We have created a separate lightweight threat model.md file and attached it as part of Appendix->References as a link

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:39]: Understood, we added links to the Vulnerability disclosure policy, default configurations, versions supported (for security updates) as well as the process that one has to go through for reporting a vulnerability.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ua2047) [reply assessments/projects/chaos-mesh/self-assessment.md:68]: Each of the actions that have associated security checks have been updated.

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (torinvdb) [inline assessments/projects/chaos-mesh/self-assessment.md:7]:
```suggestion
# Chaos Mesh Self-assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

Authors: Yash Mathur (@mathur-14), Aditya Deepak Joshi (@AdityaDJoshi), Ushan Abeysekera (@ua2047), Jia Huang (@iamgeeyuh)

Contributors/Reviewers: Pranava Kumar Vemula (@Rana-KV), Eddie Knight (@eddie-knight)

The Self-assessment is the initial document for Chaos Mesh to begin thinking about the security of the project, determining gaps in its security, and preparing any security documentation for their users.
```
@ragashreeshekar

File: assessments/projects/chaos-mesh/self-assessment.md
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:110]:
```suggestion
This self-assessment is not intended to provide a security audit of Chaos Mesh, or function as an independent assessment or attestation of Chaos Mesh's security health.
```

**Comment by ragashreeshekar** [comment]
@mathur-14 Thanks for making updates to the PR. 
Please address the pending suggestions and I think we are good to go.

**Comment by eddie-knight** [comment]
LGTM. We can merge this PR and then I'll reach out to the maintainers for next steps.

**Review by ragashreeshekar — APPROVED** [review]

**Comment by ragashreeshekar** [comment]
@mathur-14 Thanks for the updates. 
A few things to do before we could merge, please fix them:
- lint, spelling, links and lint errors
- update branch

**Comment by JustinCappos** [comment]
We don't have access to update this and merge it.  I'll close.  Please reopen if someone with the ability to edit this responds and wants to help us merge this.

**Pull request closed without merging**
